cdci_data_analysis/analysis/email_helper.py

```diff
@@ -0,0 +1,260 @@
+            'contact': 'contact@odahub.io'
+        },
+        'request': {
+            'job_id': job_id,
+            'status': status,
+            'instrument': instrument,
+            'product_type': product_type,
+            'time_request': time_request,
+            'request_url': request_url,
+            'api_code_no_token': api_code_no_token,
+        }
+    }
+
+    template = env.get_template('email.html')
+    email_body_html = template.render(**email_data)
+    
+    email_subject = re.search("<title>(.*?)</title>", email_body_html).group(1)
+
+    email_text = textify_email(email_body_html)
+    
+    server = None
+    logger.info("Sending email")
+    # Create the plain-text and HTML version of your message,
+    # since emails with HTML content might be, sometimes, not supported
+        
+    try:
+        # send the mail with the status update to the mail provided with the token
+        # eg done/failed/submitted
+        # test with the local server
+        smtp_server = config.smtp_server
+        port = config.smtp_port
+        sender_email_address = config.sender_email_address
+        cc_receivers_email_addresses = config.cc_receivers_email_addresses
+        receiver_email_address = tokenHelper.get_token_user_email_address(decoded_token)
+        receivers_email_addresses = [receiver_email_address] + cc_receivers_email_addresses
+        # creation of the message
+        message = MIMEMultipart("alternative")
+        message["Subject"] = email_subject
+        message["From"] = sender_email_address
+        message["To"] = receiver_email_address
+        message["CC"] = ", ".join(cc_receivers_email_addresses)
+
+        part1 = MIMEText(email_text, "plain")
+        part2 = MIMEText(email_body_html, "html")
+        message.attach(part1)
+        message.attach(part2)
+
+        smtp_server_password = config.smtp_server_password
+        # Create a secure SSL context
+        context = ssl.create_default_context()
+        #
+        # Try to log in to server and send email
+        server = smtplib.SMTP(smtp_server, port)
+        # just for testing purposes, not ssl is established
+        if smtp_server != "localhost":
+            try:
+                server.starttls(context=context)
+            except Exception as e:
+                logger.warning(f'unable to start TLS: {e}')
+        if smtp_server_password is not None and smtp_server_password != '':
+            server.login(sender_email_address, smtp_server_password)
+        server.sendmail(sender_email_address, receivers_email_addresses, message.as_string())
+    except Exception as e:
+        logger.error(f'Exception while sending email: {e}')
+        raise EMailNotSent(f"email not sent {e}")
+    finally:
+        if server:
+            server.quit()
+
+    store_email_info(message, status, scratch_dir)
+
+    return message
+
+
+def store_email_info(message, status, scratch_dir):
```

Comment:
This is still at the old version. I had slightly changed the way the email file name was formatted. Same for the checks

Reply:
I can also push those changes

Reply:
The refactoring changes and so have diverged with the master so it became difficult to merge. Not good the PR stayed or a whole while development progressed. 
So let's quickly get over this one?

You will make a simple change, yes?

Reply:
exacly